concurrency/runner.go

```diff
@@ -123,6 +130,8 @@ func ForEachJob(ctx context.Context, jobs int, concurrency int, jobFunc func(ctx
 
 // ForEachJobMergeResults is like ForEachJob but expects jobFunc to return a slice of results which are then
 // merged with results from all jobs. This function returns no results if an error occurred running any jobFunc.
+//
+// ForEachJobMergeResults cancels the context.Context passed to each invocation of jobFunc before ForEachJobMergeResults returns.
 func ForEachJobMergeResults[J any, R any](ctx context.Context, jobs []J, concurrency int, jobFunc func(ctx context.Context, job J) ([]R, error)) ([]R, error) {
```

Comment:
should we delete this function? We don't need it anymore and I doubt anyone was using it apart from mimir since the time it was added

Reply:
We're still using it [here](https://github.com/grafana/mimir/blob/main/pkg/distributor/distributor.go#L1508).